addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/ConnectionStateListener.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver.internal.protocol;
+
+/**
+ * Implement this to be notified of the asynchronous result of {@link ProtocolFactory}.createConnection.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public interface ConnectionStateListener {
+    public void connectionFailed(String host, Throwable throwable);
```

Comment:
Not yet. The connection part will be changed with the json protocol addition, because I need to perform a feature detection and want to do this asynchronously. That detection may fail or the connections may fail.